HHS/OPRE-OPS
# (4) 2365 upload a document: Document table

## What changed

- Create a new SQLAlchemy base model for documents.
- Add a new migration script to update the database schema.

## Issue

[Upload a document #2365](https://github.com/HHS/OPRE-OPS/issues/2365)

## How to test
1. Start and update the database using the changes: `docker compose up db data-import --build`
2. There should be a `document` table with the following columns:
<img width="336" alt="Screenshot 2024-07-17 at 5 32 17 PM" src="https://github.com/user-attachments/assets/7b5fae86-20c3-468e-92d7-875144c634fa">

3. There should be a new data type called 'documenttype' with the following enum values: 
`[CERTIFICATION_OF_FUNDING, STATEMENT_OF_REQUIREMENTS, ITAR_CHECKLIST_FOR_ALL_IT_PROCUREMENT_ACTIONS, INDEPENDENT_GOVERNMENT_COST_ESTIMATE, SECTION_508_EXCEPTION_DOCUMENTATION, COR_NOMINATION_AND_CERTIFICATION_DOCUMENT, ADDITIONAL_DOCUMENT]`

## Definition of Done Checklist


## Links

Links to prior PRs:
1. https://github.com/HHS/OPRE-OPS/pull/2406
2. https://github.com/HHS/OPRE-OPS/pull/2523
3. https://github.com/HHS/OPRE-OPS/pull/2539 
4. https://github.com/HHS/OPRE-OPS/pull/2553
